frontend/app/src/App.tsx

```diff
@@ -1525,6 +1529,20 @@ export class App extends PureComponent<Props, State> {
     this.openDeployDialog()
   }
 
+  requestFileURLs = (requestId: string, files: File[]): void => {
+    if (this.isServerConnected()) {
+      const backMsg = new BackMsg({
+        fileUrlsRequest: {
+          requestId,
+          fileNames: files.map(f => f.name),
```

Comment:
What happens if a user has two file upload widgets and uploads a file with the same name to both widgets? Does this work with the new impl. as well?

Reply:
Yes, it works correctly, because for each file we issue on the backend a different `file_id` as part of file URLs, so for the next steps from the streamlit perspective they an absolutely different files.